### PR TITLE
Dev/product categories

### PR DESCRIPTION
**Pull request by ikramabumazen**
add Categories dropDown list to POS

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -36,7 +37,7 @@ import { ProvidersModule } from './providers/providers.module';
     HttpClientModule,
     NgxDatatableModule,
     AngularFireModule.initializeApp(environment.firebase),
-    AngularFirestoreModule.enablePersistence(),, // imports firebase/firestore, only needed for database features
```
Comment (Moathomar-cp) [inline src/app/app.module.ts]: @ikramabumazen  Nice Catch, Thank you :+1:

File: src/app/categories/categories.component.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            <a class="btn btn-sm btn-default" href="#" title="Edit" data-toggle="tooltip">
+              <i class="fa fa-pencil"></i>
+            </a>
+          </div>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+  <!-- add button Add category -->
+  <button class="btn btn-add btn-lg" type="button" >Add Category</button>
```
Comment (Moathomar-cp) [inline src/app/categories/categories.component.html]: @ikramabumazen  would you mind move this button up, its not a good user experience , may be above the table is better like what @BaidaaAlbzoor  did in products list.

File: src/app/categories/categories.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (Moathomar-cp) [inline src/app/categories/categories.component.scss]: not merge blocking , but please remind to indent your files before submit , 
may be you should add an extension  to vscode, so remind me tomorrow

File: src/app/categories/categories.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  styleUrls: ['./categories.component.scss']
+})
+export class CategoriesComponent implements OnInit {
+  categoriesItems: CategoriesListItem[] = [];
+
+  constructor(private spinnerService: Ng4LoadingSpinnerService) { }
+
+  ngOnInit() {
+    //show spinner loading 
+    this.spinnerService.show();
```
Comment (Moathomar-cp) [inline src/app/categories/categories.component.ts]: @ikramabumazen  may be moving the show/hide spinner , and also the timeout into getCategoriesList is better. 
and no need for newMethod , i am sorry but its extra effort for nothing !

Comment (Moathomar-cp) [reply src/app/categories/categories.component.ts]: and newMethod name is not descriptive, i cant guess what it should do by reading it . right ?

File: src/app/core/shell/header/header.component.html [hunk 5]
```diff
@@ -12,7 +12,8 @@
       <span translate>Home</span>
     </button>
     <button mat-button routerLink="/expense" routerLinkActive="active">
-      <span translate><i class="fa fa-usd"></i> Expense</span>
+      <span translate>
+        <i class="fa fa-usd"></i> Expense</span>
```
Comment (Moathomar-cp) [inline src/app/core/shell/header/header.component.html]: :+1:

File: src/app/categories/categories.component.ts (hunk 4 above)
Comment (ikramabumazen) [reply src/app/categories/categories.component.ts]: it gives me an error on getSalesList(), fixed by using newMethod() this way

File: src/app/categories/categories.component.html (hunk 2 above)
Comment (ikramabumazen) [reply src/app/categories/categories.component.html]: Done

File: src/app/categories/categories.component.ts (hunk 4 above)
Comment (ikramabumazen) [reply src/app/categories/categories.component.ts]: fixed

**Pull request merged**